watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1556,6 +1690,170 @@ def eq_dl_thickness(self, mem, side, t, x):
                     * self.current_dens_lim_x[t, x] ** -1
                 )
 
+    def _get_fluid_dimensionless_quantities(self):
+        self.general_mass_diffusivity = Var(
+            initialize=1e-9,
+            bounds=(1e-16, 1e-6),
+            units=pyunits.meter**2 * pyunits.second**-1,
+            doc="The mass diffusivity of the solute as molecules (not individual ions)",
+        )
+        self.hydraulic_diameter = Var(initialize=1e-3, units=pyunits.meter)
+        self.Re = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Reynolds Number"
+        )
+        self.Sc = Var(
+            initialize=2000, units=pyunits.dimensionless, doc="Schmidt Number"
+        )
+        self.Sh = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Sherwood Number"
+        )
+
+        if self.config.hydraulic_diameter_method == HydraulicDiameterMethod.defined:
+            _log.warning("Do not forget to FIX the channel hydraulic diameter in [m]!")
+        else:
+
+            @self.Constraint(
+                doc="To calculate hydraulic diameter",
+            )
+            def eq_hydraulic_diameter(self):
+                if (
+                    self.config.hydraulic_diameter_method
+                    == HydraulicDiameterMethod.conventional
+                ):
+                    return (
+                        self.hydraulic_diameter
+                        == 2
+                        * self.channel_height
+                        * self.cell_width
+                        * self.spacer_porosity
+                        * (self.channel_height + self.cell_width) ** -1
```

Comment:
I think this _might_ be incorrect. Hydraulic diameter is 4* cross-sectional area divided by the wetted perimeter, accounting for the spacer. It looks like you're account for spacer porosity for the area but not the wetted perimeter.

Reply:
For a secondary reference, you can look at how I incorporated `dh` for the RO model.

Reply:
Get your point Adam.  I carefully reviewed literature on this and actually had the same question to yours- but this equation was used by a few papers. So my take on this is that the hydraulic diameter for a porous media gonna carry a higher degree of error/uncertainty than regular channels (e.g., pipes) so any equation can only do so much.  The other method I had is corresponding to the method in RO with a modification to reflect the original derivation of that equation.  I can explain this more in tomorrow's meeting.

Reply:
Upon some discussion with Tim, decisions related to friction factor and hydraulic diameter methods are (1) taking the method needing a `express_if` out to avoid complicatedness, (2) keep the specific surface area (S_v) d_h method as a more general equation and I can note somewhere (e.g., documentation) that under certain circumstances S_v can be approximated by 8/h, which leads to the eq used in RO.